screenpy_selenium/common.py

```diff
@@ -0,0 +1,45 @@
+"""module to hold shared objects."""
+
+from __future__ import annotations
+
+import warnings
+from functools import wraps
+from typing import TYPE_CHECKING, Callable, TypeVar
+
+from typing_extensions import ParamSpec
+
+if TYPE_CHECKING:
+    P = ParamSpec("P")
+    T = TypeVar("T")
+    Function = Callable[P, T]
+
+
+def pos_args_deprecated(*keywords: str) -> Function:
+    """Warn users which positional arguments should be called via keyword."""
+
+    def deprecated(func: Function) -> Function:
+        argnames = func.__code__.co_varnames[: func.__code__.co_argcount]
+        i = min([argnames.index(kw) for kw in keywords])
+        kw_argnames = argnames[i:]
+
+        @wraps(func)
+        def wrapper(*args: P.args, **kwargs: P.kwargs) -> Function:
+            # call the function first, to make sure the signature matches
+            ret_value = func(*args, **kwargs)
+
+            args_that_should_be_kw = args[i:]
+            if args_that_should_be_kw:
+                posargnames = ", ".join(kw_argnames)
+
+                msg = (
+                    f"Warning: positional arguments `{posargnames}` for "
+                    f"`{func.__qualname__}` are deprecated.  "
+                    f"Please use keyword arguments instead."
```

Comment:
Do we want to mention what version we'll be making this a concrete change? Version 5?

Reply:
I think that would be a good idea.  Does 5 make sense?   I don't have a good sense for what kind of timeline that entails.